src/npcmove.cpp

```diff
@@ -3772,16 +3800,14 @@ static float rate_food( const item &it, int want_nutr, int want_quench )
     }
 
     if( nutr > want_nutr ) {
-        // TODO: Allow overeating in some cases
-        if( nutr >= 5 ) {
-            return 0.0f;
-        }
```

Comment:
I've removed this because 1 nutr is a bit under 10kCal, and refusing to eat a food when hungry because it's less than 50kCal over seems like *very* specific calorie counting.

It would also prevent NPCs drinking a soda or sports drink to relieve thirst if they were thirsty but not hungry.

Honestly I'd love to adjust (and potentially cache) `rate_food()`, but I'd prefer to do that as a separate PR.

Reply:
Agreed with removing the check. Not sure if `want_nutr` can be negative, if not it might not have been totally wrong though.
The weight of overeating penalty should be reduced, I think. Now it massively influences calculations.